osf/models/preprint_service.py

```diff
@@ -206,5 +206,7 @@ def set_preprint_license(self, license_detail, auth, save=False):
 
     def save(self, *args, **kwargs):
         saved_fields = super(PreprintService, self).save(*args, **kwargs)
-        if saved_fields:
+        if not saved_fields:
+            saved_fields = []
+        if 'is_published' in saved_fields or self.is_published:
```

Comment:
`saved_fields` will always be an empty list here; Django doesn't return saved fields like MODM did. See `DirtyFieldsMixin`